[dynamo] bugfix - make module setattr more restrictive

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107742
* __->__ #107828

A check got missed in https://github.com/pytorch/pytorch/pull/106092

Fixes https://github.com/pytorch/pytorch/issues/107721

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

anijain2305 committed:
[dynamo] bugfix - make module setattr more restrictive

[ghstack-poisoned]

eellison reviewed (approved):
thanks, sorry abt that!

anijain2305 commented:
@pytorchbot merge

anijain2305 commented:
@pytorchbot merge

anijain2305 committed:
Update on "[dynamo] bugfix - make module setattr more restrictive"


A check got missed in https://github.com/pytorch/pytorch/pull/106092

Fixes https://github.com/pytorch/pytorch/issues/107721

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] bugfix - make module setattr more restrictive"


A check got missed in https://github.com/pytorch/pytorch/pull/106092

Fixes https://github.com/pytorch/pytorch/issues/107721

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge

anijain2305 commented:
@pytorchbot merge -f "expecting ROCm tests to pass"
